src/transformStateWithClones.js

```diff
@@ -7,7 +7,34 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const actionState = { ...state };
+  const history = [];
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(actionState, action.extraData);
+        break;
+
+      case 'removeProperties':
+        for (const keyToRemove of action.keysToRemove) {
+          delete actionState[keyToRemove];
+        }
+        break;
+
+      case 'clear':
+        for (const key in actionState) {
+          delete actionState[key];
```

Comment:
Keep one way of naming

Reply:
I`m sorry, but I don`t think this is necessary